pkg/build/controller/strategy/util.go

```diff
@@ -90,3 +65,49 @@ func setupBuildEnv(build *buildapi.Build, pod *kapi.Pod) error {
 	}
 	return nil
 }
+
+// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
+// allowing Docker to authenticate against private registries or Docker Hub.
+func setupDockerSecrets(pod *kapi.Pod, secretRef *kapi.ObjectReference) {
+	if secretRef == nil {
+		return
+	}
+
+	// TODO: Should namespace be required here?
+	namespace := secretRef.Namespace
+	if len(namespace) == 0 {
+		namespace = pod.Namespace
+	}
+
+	if len(namespace) == 0 {
+		glog.Infof("Unable to get namespace from RegistrySecret, using default")
+		namespace = kapi.NamespaceDefault
+	}
+
+	volume := kapi.Volume{
+		Name: secretRef.Name,
+		VolumeSource: kapi.VolumeSource{
+			Secret: &kapi.SecretVolumeSource{
+				Target: kapi.ObjectReference{
+					Kind:      "Secret",
+					Name:      secretRef.Name,
+					Namespace: namespace,
+				},
+			},
+		},
+	}
+	volumeMount := kapi.VolumeMount{
+		Name:      secretRef.Name,
+		MountPath: dockerSecretMountPath,
+		ReadOnly:  true,
+	}
+
+	glog.V(3).Infof("Adding volume with %s to Build pod %s", secretRef.Name, pod.Name)
+
+	pod.Spec.Volumes = append(pod.Spec.Volumes, volume)
+	pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, volumeMount)
+	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, kapi.EnvVar{
+		Name:  "DOCKERCFG_PATH",
+		Value: filepath.Join(volumeMount.MountPath, secretRef.Name),
```

Comment:
Path to file. It is read by the builder.
On Mar 23, 2015 7:41 PM, "Jordan Liggitt" <notifications@github.com> wrote:

> In pkg/build/controller/strategy/util.go
> <https://github.com/openshift/origin/pull/1411#discussion_r26966124>:
>
> > +			},
> > +		},
> > +	}
> > +	volumeMount := kapi.VolumeMount{
> > +		Name:      secretRef.Name,
> > +		MountPath: dockerSecretMountPath,
> > +		ReadOnly:  true,
> > +	}
> > +
> > +	glog.V(3).Infof("Adding volume with %s to Build pod %s", secretRef.Name, pod.Name)
> > +
> > +	pod.Spec.Volumes = append(pod.Spec.Volumes, volume)
> > +	pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, volumeMount)
> > +	pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, kapi.EnvVar{
> > +		Name:  "DOCKERCFG_PATH",
> > +		Value: filepath.Join(volumeMount.MountPath, secretRef.Name),
>
> is this supposed to be the path to a directory or the dockercfg file
> itself?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1411/files#r26966124>.
>